test/sys/test_inotify.rs

```diff
@@ -0,0 +1,54 @@
+use nix::sys::inotify::{EventFlags,InotifyInitFlags,inotify_init1,inotify_add_watch};
+use nix::sys::inotify::ffi::inotify_event;
+use nix::Error;
+use nix::errno::Errno;
+use nix::unistd::read;
+use nix::sys::stat::Mode;
+use tempfile;
+use std::fs::OpenOptions;
+use std::mem;
+use std::io::Write;
+use std::os::unix::fs::OpenOptionsExt;
+
+#[test]
+pub fn test_inotify() {
+	let ifd = inotify_init1(InotifyInitFlags::IN_NONBLOCK).unwrap();
+	let tempdir = tempfile::tempdir().unwrap();
+
+	let wd = inotify_add_watch(ifd, tempdir.path(), EventFlags::IN_ALL_EVENTS);
+	assert!(wd.is_ok());
+
+	let mut buffer = [0u8; 4096];
+
+	let mut _nread = read(ifd, &mut buffer);
```

Comment:
Why the leading underscore on `_nread`?  That usually indicates a variable that will not be used.